### ISSUE TITLE
Dependency versions are incorrect - everything is ^v2024.10.1

### ISSUE THREAD

**Issue by orf** [metio/kube-custom-resources-rs#345]
Hey,
The dependencies for the currently published crate are incorrect. See: https://crates.io/crates/kube-custom-resources-rs/2024.10.1/dependencies

It seems that all the dependency versions are being replaced with the crate version?

This is only present in the `v2024.10.1` release: `v2024.9.1` has the correct dependencies.

![Screenshot 2024-10-24 at 11 28 19](https://github.com/user-attachments/assets/e62eb1c0-e351-4c1d-b0b0-f027507c5a6a)

**Comment by sebhoss** [comment]
yeah sorry I broke this while working on https://github.com/metio/kube-custom-resources-rs/issues/335 - I'm going to fix this and create a new release asap.

**Comment by sebhoss** [comment]
I cannot currently publish a release because this crate is now too large for crates.io (see https://github.com/metio/kube-custom-resources-rs/actions/runs/11504507909/job/32024141108). I've contacted the crates.io team to see whether they can increase the limit for this crate.

Sorry again for the inconvenience!

**Comment by orf** [comment]
No need to apologize - thank you for this crate!

one alternative to sidestep the size issues might be to publish each source individually as a separate crate? That would also allow releasing sub packages as they are updated/independently.

I could have a look at this if you’re interested?

**Comment by sebhoss** [comment]
Yes I'm very interested in that!

I previously reduced the release frequency because the crates team asked me to after increasing the limit of allowed features per crate which I hit last year 😅

Splitting into multiple smaller crates might be beneficial for that as well since each small crate would only declare a handful of features and thus it would be fine to release frequently changing CRDs sooner while releasing those that rarely change less frequently.

If you want to give it a shot, take a look at the code-generator folder. It contains scripts and binaries that download CRDs and move the generated code to its target destination. Currently, they all end up in that one crate, but we could split by group maybe? Or even group/version?

**Comment by orf** [comment]
Ok! I took a look this evening. The generation scripts had some issues on my machine, so I started to add part of the functionality to the `crd_fetcher_v1` file as an exercise to investigate how everything fits together.

I'll give the actual splitting into multiple crates bit a go this week, but let me know if you want me to continue with moving some of the shell script logic into `crd_fetcher_v1`. Absolutely no worries if not - it was just an experiment while I learn, but if you do like the idea then I can carry on in that direction.

Here's the branch so far, if you're interested. I've done up to `create-mod-rs-files.sh`.

https://github.com/metio/kube-custom-resources-rs/compare/main...orf:kube-custom-resources-rs:refactor
